gmail/modules/listener/listener.bal

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerinax/googleapis_gmail as gmail;
+
+# Listener for Gmail Connector.
+@display {label: "Gmail Listener"} 
+public class Listener {
+    private string startHistoryId = "";
+    private string topicResource = "";
+    private string subscriptionResource = "";    
+    private string userId = ME;
+    private gmail:Client gmailClient;
+    private http:Listener httpListener;
+    private string project;
+    private string pushEndpoint;
+
+    private json requestBody;
+    private HttpService httpService;
+    http:Client pubSubClient;
+
+
+    public isolated function init(int port, gmail:GmailConfiguration gmailConfig, string project, string pushEndpoint) 
+                                  returns @tainted error? {
+
+        http:ClientSecureSocket? socketConfig = gmailConfig?.secureSocketConfig;
+        // Create pubsub http client.
+        if (socketConfig is http:ClientSecureSocket) {
+            self.pubSubClient = checkpanic new (PUBSUB_BASE_URL, {
```

Comment:
What is the reason for `checkpanic`? Please provide PR or commit if this has been addressed in the master.

Reply:
This is corrected here : https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/commit/5a1d71c7f012e587bed57a1e5d58905d006ef3fa#diff-b32b4357eb06cb8b495e1444316e33ecda4c8c466da98128af02da742264faffR52